Gems/ScriptCanvas/Code/Editor/View/Windows/MainWindow.cpp

```diff
@@ -1656,41 +1659,47 @@ namespace ScriptCanvasEditor
             }
         }
 
-        if (inMemoryAssetId.Path().Extension() == ".scriptevents")
+        if (sourceHandle.Path().Extension() == ".scriptevents")
         {
-            auto newPath = inMemoryAssetId.Path();
+            auto newPath = sourceHandle.Path();
             newPath.ReplaceExtension(".scriptcanvas");
-            inMemoryAssetId = SourceHandle(inMemoryAssetId, newPath);
+            if (auto relativeOption = ScriptCanvasEditor::CreateFromAnyPath(sourceHandle, newPath))
+            {
+                sourceHandle = *relativeOption;
+            }
         }        
 
-        if (!m_activeGraph.AnyEquals(inMemoryAssetId))
+        if (!m_activeGraph.AnyEquals(sourceHandle))
         {
-            OnChangeActiveGraphTab(inMemoryAssetId);
+            OnChangeActiveGraphTab(sourceHandle);
         }
 
-        PrepareAssetForSave(inMemoryAssetId);
+        PrepareAssetForSave(sourceHandle);
 
         AZStd::string suggestedFilename;
         AZStd::string suggestedFileFilter;
         bool isValidFileName = false;
-            
+
+        AZ::IO::FixedMaxPath projectSourcePath = AZ::Utils::GetProjectPath();
+        projectSourcePath /= "ScriptCanvas";
+
         if (save == Save::InPlace)
         {
             isValidFileName = true;
             suggestedFileFilter = SourceDescription::GetFileExtension();
-            suggestedFilename = inMemoryAssetId.Path().c_str();
+            suggestedFilename = sourceHandle.Path().c_str();
         }
         else
         {
             suggestedFileFilter = SourceDescription::GetFileExtension();
 
-            if (inMemoryAssetId.Path().empty())
+            if (sourceHandle.Path().empty())
             {
-                suggestedFilename = SourceDescription::GetSuggestedSavePath();
+                suggestedFilename = projectSourcePath.Native();
             }
             else
             {
-                suggestedFilename = inMemoryAssetId.Path().c_str();
+                suggestedFilename = sourceHandle.Path().c_str();
```

Comment:
repeat: `.Native()` can be used here to return an `AZStd::string&` instead of a `const char*` which results in a slightly quicker assignment(strlen doesn't need to be calculated).